core/src/jvmMain/kotlin/com.saveourtool.processbuilder/ProcessBuilderInternal.kt

```diff
@@ -86,11 +88,12 @@ actual class ProcessBuilderInternal actual constructor(
                 }
             )
         )
-        val data = br.useLines {
-            it.joinToString("\n")
-        }
-        fs.write(file) {
-            write(data.encodeToByteArray())
+        val data = br.readLines()
+        file?.let {
```

Comment:
On JVM, you need at least two threads to monitor a process for output w/o risking thread starvation. And I'm not even sure what should we do for K/N. Let's discuss this privately.